src/directives/badge/badge.directive.ts

```diff
@@ -0,0 +1,207 @@
+        }
+
+        if (this._badgeBorderColor) {
+            this._renderer.setStyle(badgeElement, 'border-color', this._badgeBorderColor.toHex());
+        }
+
+        this._badgeElement = badgeElement;
+        this._renderer.appendChild(this._element.nativeElement, this._badgeElement);
+        this._renderer.removeStyle(this._badgeElement, 'display');
+    }
+
+    ngOnChanges(changes: SimpleChanges): void {
+        let contentChanged: boolean = false;
+        // get display friendly version of text based on max length and type of val;
+        if (changes.badgeContent || changes.badgeMaxValue) {
+            let finalText: string =
+                (changes.badgeContent && changes.badgeContent.currentValue) || this.badgeContent || null;
+            const maxValue: number =
+                (changes.badgeMaxValue && changes.badgeMaxValue.currentValue) || this.badgeMaxValue || null;
+
+            if (finalText && maxValue && maxValue > 0) {
+                if (this._isNumber && parseInt(finalText) > maxValue) {
+                    finalText = `${maxValue}+`;
+                } else if (finalText.length > maxValue) {
+                    finalText = `${finalText.substr(0, maxValue)}&hellip;`;
+                }
+            }
+
+            this._badgeDisplayContent = finalText;
+            contentChanged = true;
+        }
+
+        // if the badge is visible set changed values
+        if (this._badgeElement) {
+            // set the badge content
+            if (contentChanged) {
+                this._renderer.setProperty(this._badgeElement, 'innerHTML', this._badgeDisplayContent);
+            }
+
+            // set the badge color
+            if (changes.badgeColor && changes.badgeColor.currentValue !== changes.badgeColor.previousValue) {
+                this._renderer.setStyle(this._badgeElement, 'background', this._badgeColor.toHex());
+                this._renderer.setStyle(this._badgeElement, 'color', this.determineContentTextColor().toHex());
+            }
+
+            // set the badge border color
+            if (changes.badgeBorderColor && changes.badgeBorderColor.currentValue !== changes.badgeBorderColor.previousValue) {
+                this._renderer.setStyle(this._badgeElement, 'border-color', this._badgeBorderColor.toHex());
+            }
+        }
+    }
+
+    ngOnDestroy(): void {
+        if (this._renderer.destroyNode) {
+            this._renderer.destroyNode(this._badgeElement);
+        }
+    }
+
+    private determineContentTextColor(): ThemeColor {
+        return this._badgeColor
+            ? ThemeColor.parse(
+                this._contrastService.getContrastColor(this._badgeColor, this._lightColor, this._darkColor).toRgba()
+            )
+            : this._lightColor;
+    }
+
+    private parseThemeColor(color: string): ThemeColor {
+        let themeColor: ThemeColor = null;
+
+        if (color) {
```

Comment:
We would usually write this as
```ts
return color ? ThemeColor.parse(this._colorService.resolve(color)) : null;
```

Reply:
@donna-marijne I had to add a check that the colour is valid or it errors out - but this is directly related to typing in a new colour in the docs.
```suggestion
       return color && this._colorService.colorExists(color) ? ThemeColor.parse(this._colorService.resolve(color)) : null;
```